gradio/blocks.py

```diff
@@ -2616,13 +2616,16 @@ def block_thread(
         """Block main thread until interrupted by user."""
         try:
             while True:
-                time.sleep(0.1)
+                if self.is_running:
```

Comment:
HI @DoiiarX ! Can you please provide some more information about what this is solving? I can close running gradio applications with ctrl + C with the code as it is.

Reply:
Nevermind just saw the linked issue. Will take a look.

Reply:
Ok took a look and I'm still confused. When I run the linked code, the server does not restart.

<img width="642" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/0192d6a0-3baf-4644-b923-350c806cf0c7">

Starting gradio outside the main thread is not really possible because the asyncio event loop is only started in the main thread. So I think it does not make sense to support this use case.